core/sql/exp/exp_datetime.cpp

```diff
@@ -2019,6 +2051,14 @@ ExpDatetime::convAsciiToDatetime(char *srcData,
   //
   if (format == DATETIME_FORMAT_ERROR) {
     ExRaiseSqlError(heap, diagsArea, EXE_CONVERT_DATETIME_ERROR);
+    char hexstr[256];
+    memset(hexstr, 0 , sizeof(hexstr) );
```

Comment:
I wonder if lines 2055 to 2059 is equivalent to the following two lines
strncpy(hexstr,originalSrcData,255);  // (using constant declaration-1 instead of 255 of course :))
hexstr[255] = 0;

I guess I am trying to say that strncpy is capable of handling source strings larger than destination and 2057 and 2058 seem unnecessary.